torch/fft/__init__.py

```diff
@@ -9,18 +9,283 @@
 # connects the torch.fft Python namespace to the torch._C._fft builtins.
 
 fft = _add_docstr(_fft.fft_fft, r"""
-fft(input) -> Tensor
+fft(input, n=None, dim=-1, norm=None) -> Tensor
 
 Computes the one dimensional discrete Fourier transform of :attr:`input`.
 
+Note:
+
+    The Fourier domain representation of any real signal satisfies the
+    Hermitian property: `X[i] = conj(X[-i])`. This function always returns both
+    the positive and negative frequency terms, even though for real inputs the
+    negative frequencies are redundant. :func:`~torch.fft.rfft` returns the more
+    efficient "half-complex" representation where only the positive frequencies
```

Comment:
This is a reasonable description of "half-complex," but readers of the C++ docstrings may not encounter it.